src/njoy21/input/PLOTR/test/PLOTR.test.cpp

```diff
@@ -0,0 +1,779 @@
+      REQUIRE( std::nullopt == plotr.curves[0].second.card10a );
+      REQUIRE( std::nullopt == plotr.curves[0].second.freeform );
+    }
+  }//WHEN
+// Freeform input
+//std::string sCard4_3( "1 /\n" );
+//std::string sCard5_3( " /\n" );
+//std::string sCard5a_3( " /\n" );
+//std::string sCard6_3( " /\n" );
+//std::string sCard6a_3( " /\n" );
+// No card 7 or 7a.
+//std::string sCard8_3( " 0 /\n" );
+// No card 10 or 10a or 11.
+//std::string sCard12_3( " 0 /\n" );
+//std::string sCard13_3_1( " 1.0 2.0 1.5 2.5 0.5 1.5 /\n" );
+//std::string sCard13_3_2( " 5.0 5.0 0 0 0 0 /\n" );
+//std::string sCard13_3_3( " 20.1 21.2 10.0 30.0 10.0 25.0 /\n" );
+//std::string sCard13_3_4( " /\n" );
+
+  WHEN( "A valid freeform input" ){
+    iRecordStream<char> iss( std::istringstream(
+      sCard0
+      + sCard1
+      + sCard2_1
+      + " /\n" // empty title
+      + " /\n" // empty title
+      + sCard4_3 // 1 0 2 0
+      + sCard5_3
+      + sCard5a_3
+      + sCard6_3
+      + sCard6a_3
+      + sCard8_3
+      + sCard9_1
+      + sCard12_3
+      + sCard13_3_1
+      + sCard13_3_2
+      + sCard13_3_3
+      + sCard13_3_4
+      + sCard2_2
+    ) );
+
+    PLOTR plotr( iss );
+
+    THEN( "the Card0 input values can be verified: 23" ){
+      REQUIRE( 23 == plotr.card0.nplt.value );
+      REQUIRE( 0  == plotr.card0.nplt0.value );
+    }
+    THEN( "the Card1 input values can be verified: 1 1 0.1 1" ){
+      REQUIRE( 1 == plotr.card1.lori.value );
+      REQUIRE( 1 == plotr.card1.istyle.value );
+      REQUIRE( Approx( 0.1 ) == plotr.card1.size.value );
+      REQUIRE( 1 == plotr.card1.ipcol.value );
+    }
+    THEN( "the curves variable should have a length of 1" ){
+      REQUIRE( 1 == plotr.curves.size() );
+    }
+    THEN(
+    "the Card2 input values can be verified: 1 7 1.1 2.1 1.2 1.3 5.0 5.1 0.1" ){
+      REQUIRE( 1 == plotr.curves[0].first.iplot.value );
+      REQUIRE( 7 == plotr.curves[0].first.iwcol.value );
+      REQUIRE( Approx( 1.1 ) == plotr.curves[0].first.factx.value );
+      REQUIRE( Approx( 2.1 ) == plotr.curves[0].first.facty.value );
+      REQUIRE( Approx( 1.2 ) == plotr.curves[0].first.xll.value );
+      REQUIRE( Approx( 1.3 ) == plotr.curves[0].first.yll.value );
+      REQUIRE( Approx( 5.0 ) == plotr.curves[0].first.ww.value );
+      REQUIRE( Approx( 5.1 ) == plotr.curves[0].first.wh.value );
+      REQUIRE( Approx( 0.1 ) == plotr.curves[0].first.wr.value );
+    }
+    THEN( "the Card3 input values can be verified" ){
+      REQUIRE( PLOTR::Card3::Title::defaultValue() == 
+               plotr.curves[0].second.plot->card3.title.value );
+    }
+    THEN( "the Card3 input values can be verified" ){
+      REQUIRE( PLOTR::Card3a::Title::defaultValue() == 
+               plotr.curves[0].second.plot->card3a.title.value );
+    }
+    THEN( "the Card4 input values can be verified: 1" ){
+      REQUIRE( 1 == plotr.curves[0].second.plot->card4.itype.value );
```

Comment:
Stating something like:

     auto curve = plot.curves[0].second

could make this section much more compact and easier to read. A similar thing can be done above with `plot.curves[0].first`.